File: proyecto-veterinaria-laravel/app/Models/Cita.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Cita extends Model
{
    use HasFactory;

    protected $table = 'citas';

    protected $fillable = [
        'usuario_id',
        'mascota_id',
        'veterinario_id',
        'nombre_dueno',
        'email',
        'telefono',
        'nombre_mascota', // Se mantiene por legacy o si no tiene mascota registrada
        'tipo_mascota',
        'tipo_servicio',
        'fecha_preferida',
        'hora_preferida',
        'notas',
        'estado',
        'diagnostico',
        'tratamiento',
        'notas_internas'
    ];

    public function usuario()
    {
        return $this->belongsTo(Usuario::class);
    }

    public function mascota()
    {
        return $this->belongsTo(Mascota::class);
    }

    public function veterinario()
    {
        return $this->belongsTo(Usuario::class, 'veterinario_id');
    }
}

File: maskotas/src/main/java/com/maskotas/model/Cita.java
package com.maskotas.model;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDate;

@Entity
@Table(name = "citas")
@Data
public class Cita {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String tipoServicio;
    private LocalDate fechaPreferida;
    private String horaPreferida;
    private String notas;

    private String nombreDueno;
    private String email;
    private String telefono;
    private String nombreMascota;
    private String tipoMascota;

    @Enumerated(EnumType.STRING)
    private EstadoCita estado;

    private String diagnostico;
    private String tratamiento;
    private String notasInternas;

    @ManyToOne
    @JoinColumn(name = "usuario_id")
    private Usuario usuario;

    @ManyToOne
    @JoinColumn(name = "mascota_id")
    private Mascota mascota;

    @ManyToOne
    @JoinColumn(name = "veterinario_id")
    private Usuario veterinario;
}
